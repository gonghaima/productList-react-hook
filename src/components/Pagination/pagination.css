.paginationContainer {
  display: flex;
  flex-direction: row-reverse;
}

.pagination {
  display: inline-block;
  padding-left: 15px;
  padding-right: 15px;
}

.pagination li {
  display: inline-block;
}

.pagination a {
  z-index: 3;
  color: var(--general-anchor-color);
  cursor: pointer;
  position: relative;
  float: left;
  padding: 6px 12px;
  margin-left: -1px;
  line-height: 1.42857143;
  text-decoration: none;
  outline: unset;
  font-size: var(--font-size-xs);
}

.active a {
  background: var(--active-anchor-color);
  outline: unset;
  border-bottom: 3px solid var(--general-anchor-color);
}
.disabled a,
.disabled a:hover {
  color: var(--disabled-anchor-color);
  cursor: default;
  outline: none;
  font-weight: unset;
  font-size: var(--font-size-sm);
}

.pagination a:hover {
  font-size: var(--font-size-m);
  font-weight: bold;
}

@media (min-width: 768px) {
  .pagination a {
    font-size: var(--font-size-sm);
  }
}
