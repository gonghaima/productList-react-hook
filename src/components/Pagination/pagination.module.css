.general {
  display: flex;
  flex-direction: row-reverse;
}
