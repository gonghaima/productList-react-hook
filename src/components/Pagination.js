import React from "react";
import { withRouter } from "react-router-dom";
import ReactPaginate from "react-paginate";
import { Store } from "../store";
import { fetchDataAction } from "../actions";
import footerStyles from "../footer.module.css";
import pagination from "../reducers/pagination";

const Pagination = ({ history }) => {
  const { state, dispatch } = React.useContext(Store);
  const { apiUrl, currentLimit, currentOffset } = state;

  return (
    <div className={footerStyles.general}>
      <ReactPaginate
        previousLabel={"< Previous page"}
        nextLabel={"Next page >"}
        breakLabel="..."
        breakClassName="break-me"
        marginPagesDisplayed={state.marginPagesDisplayed}
        pageCount={state.pageCount}
        pageRangeDisplayed={state.pageRangeDisplayed}
        forcePage={state.currentOffset}
        onPageChange={e =>
          fetchDataAction(dispatch, apiUrl, e.selected, currentLimit, history)
        }
        containerClassName="pagination"
        subContainerClassName="pages pagination"
        activeClassName="active"
      />
    </div>
  );
};

export default withRouter(Pagination);
