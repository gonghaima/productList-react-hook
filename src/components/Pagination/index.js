import React from "react";
import { withRouter } from "react-router-dom";
import ReactPaginate from "react-paginate";
import { Store } from "../../store";
import { fetchDataAction } from "../../actions";
import "./pagination.css";

const Pagination = ({ history }) => {
  const { state, dispatch } = React.useContext(Store);
  const { apiUrl, currentLimit, currentOffset, marginPagesDisplayed, pageRangeDisplayed, pageCount } = state;

  return (
    <div className="paginationContainer">
      <ReactPaginate
        previousLabel={"< Previous page"}
        nextLabel={"Next page >"}
        breakLabel="..."
        breakClassName="break-me"
        marginPagesDisplayed={marginPagesDisplayed}
        pageCount={pageCount}
        pageRangeDisplayed={pageRangeDisplayed}
        forcePage={currentOffset}
        onPageChange={e =>
          fetchDataAction(dispatch, apiUrl, e.selected, currentLimit, history)
        }
        containerClassName="pagination"
        subContainerClassName="pages pagination"
        activeClassName="active"
      />
    </div>
  );
};

export default withRouter(Pagination);
