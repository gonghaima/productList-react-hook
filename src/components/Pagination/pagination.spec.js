import React from "react";
import { shallow } from "enzyme";
import Pagination from "./index";

it("Pagination component renders without crashing", () => {
  const PaginationComponent = shallow(<Pagination />);
  expect(PaginationComponent.debug()).toMatchSnapshot();
});
