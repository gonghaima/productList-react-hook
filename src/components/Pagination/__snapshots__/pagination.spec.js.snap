// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component renders without crashing 1`] = `
"<ContextConsumer>
  [function]
</ContextConsumer>"
`;
